platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -150,10 +313,15 @@ protected void writeLongsData(final FileChannel fc) {}
     @Override
     protected long lookupInChunk(final long chunkIndex, final long subIndex) {
         try {
-            final ByteBuffer buf = TEMP_LONG_BUFFER_THREAD_LOCAL.get();
-            final long offset = FILE_HEADER_SIZE + (((chunkIndex * numLongsPerChunk) + subIndex) * Long.BYTES);
+            final ByteBuffer buf = tempLongBufferThreadLocal.get();
```

Comment:
Move it to where it's used to avoid wasting it in case you return due to the condition below.

Reply:
It's not exactly wasted. It's being initialized only once. But point taken, I'll move it below the `if`